fireplace/actions.py

```diff
@@ -705,9 +707,22 @@ class Discover(TargetedAction):
 	ARGS = ("TARGET", "CARDS")
 
 	def get_target_args(self, source, target):
+		if target.hero.data.card_class != CardClass.NEUTRAL:
+			# use hero class for Discover if not neutral (eg. Ragnaros)
+			discoverclass = target.hero.data.card_class
+		elif source.data.card_class != CardClass.NEUTRAL:
+			# use card class for neutral hero classes
+			discoverclass = source.data.card_class
+		else:
+			# use random class for neutral hero classes with neutral cards
+			discoverclass = cards.db[random.choice(cards.filter(collectible=True, type=CardType.HERO))].card_class
```

Comment:
This is super slow and nasty anyway. Instead, just pick from a random CardClass, from a list. write a separate utils function from it (random_class).